librosa/display.py

```diff
@@ -1220,8 +1323,197 @@ def __coord_time(n, sr=22050, hop_length=512, **_kwargs):
     """Get time coordinates from frames"""
     return core.frames_to_time(np.arange(n + 1), sr=sr, hop_length=hop_length)
 
+
 def __same_axes(x_axis, y_axis, xlim, ylim):
     """Check if two axes are the same, used to determine squared plots"""
     axes_same_and_not_none = (x_axis == y_axis) and (x_axis is not None)
     axes_same_lim = xlim == ylim
     return axes_same_and_not_none and axes_same_lim
+
+
+def waveshow(
+    y,
+    sr=22050,
+    max_points=11025,
+    x_axis="time",
+    offset=0.0,
+    marker="",
+    where="post",
+    label=None,
+    ax=None,
+    **kwargs,
+):
+    """Visualize a waveform in the time domain.
+
+    This function constructs a plot which adaptively switches between a raw
+    samples-based view of the signal (`matplotlib.pyplot.step`) and an
+    amplitude-envelope view of the signal (`matplotlib.pyplot.fill_between`)
+    depending on the time extent of the plot's viewport.
+
+    More specifically, when the plot spans a time interval of less than `max_points /
+    sr` (by default, 1/2 second), the samples-based view is used, and otherwise a
+    downsampled amplitude envelope is used. 
+    This is done to limit the complexity of the visual elements to guarantee an
+    efficient, visually interpretable plot.
+
+    When using interactive rendering (e.g., in a Jupyter notebook or IPython
+    console), the plot will automatically update as the view-port is changed, either
+    through widget controls or programmatic updates.
+
+
+    Parameters
+    ----------
+    y : np.ndarray [shape=(n,) or (2,n)]
+        audio time series (mono or stereo)
+
+    sr : number > 0 [scalar]
+        sampling rate of ``y``
+
+    max_points : postive integer
+        Maximum number of samples to draw.  When the plot covers a time extent
+        smaller than `max_points / sr` (default: 1/2 second), samples are drawn.
```

Comment:
"all samples are drawn"?

Reply:
No, just the samples within the viewport.

Reply:
ah, ok, i understand the sentence now. the when should read as an "iff"